daisy/step_create_images.go

```diff
@@ -16,63 +16,124 @@ package daisy
 
 import (
 	"context"
+	"encoding/json"
 	"sync"
 
 	"google.golang.org/api/googleapi"
 )
 
 // CreateImages is a Daisy CreateImages workflow step.
-type CreateImages []*Image
+type CreateImages struct {
+	Images     []*Image
+	ImagesBeta []*ImageBeta
+}
+
+// UnmarshalJSON unmarshals Image.
+func (ci *CreateImages) UnmarshalJSON(b []byte) error {
+	imagesBeta := &[]*ImageBeta{}
```

Comment:
Dont take the reference here so you don't need to dereference later, also this is more idiomatic:

var imagesBeta []*ImageBeta
if err := json.Unmarshal(b, &imagesBeta); err != nil {
		return err
	}
ci.ImagesBeta = imagesBeta

Reply:
What do you exactly mean with "dont take the reference here"?

Reply:
Meant to say take the address, when you instantiate imagesBeta you take the address of a slice that you then deference in line 37.

Reply:
Wouldn't that copy images/imagesBeta objects when assigning to ci.Images/ciImagesBeta?

Reply:
Updated code after chatting in person